### PR TITLE
Greg login/authentication

### PR DESCRIPTION
**Pull request by gneagu**
Merging authentication

### CODE REVIEW

**Comment by gneagu** [comment]
Hold on. I forgot some content.

File: socialdistribution/profiles/views.py [hunk 1]
```diff
@@ -45,17 +63,20 @@ def author_posts(request, author_id):
     return HttpResponse("Here are the posts of %s: ", author_id)
 
 def view_profile(request):
-    author = Author.objects.get(displayName= 'Xiaole')
+    author = Author.objects.get(displayName= 'Xiaole') #hardcode here
```
Comment (XiaoleZ) [inline socialdistribution/profiles/views.py]: maybe we all need to change to `Author.objects.get( id ="...")

File: socialdistribution/profiles/views.py [hunk 2]
```diff
@@ -69,3 +90,17 @@ def edit_profile(request):
...
+
+def register(request):
+    if request.method == "POST":
+        form = ProfileSignup(request.POST)
+        print("BEFORE")
+        if form.is_valid():
+            print("FORM VALID")
+            form.save()
+            return redirect("/login")
+        return redirect("/posts")
```
Comment (XiaoleZ) [inline socialdistribution/profiles/views.py]: right now we use `hostname/stream` to display all the post.

File: socialdistribution/profiles/views.py (hunk 1 above)
Comment (gneagu) [reply socialdistribution/profiles/views.py]: Yes. It's changed in a new commit that I messed up. Just waiting for this to be merged so I can branch again, and copy over code to fix.

**Pull request merged**